### PR TITLE
[x264] Fix *inx dynamic build

### LINKED ISSUE #13683: [x264] x64-linux-dynamic install failure
**Host Environment**
- OS: Linux

**Failure logs** 
When building shared library on linux with modified triplet: set(VCPKG_LIBRARY_LINKAGE dynamic), installation fails with:
```
-- Configuring x64-linux-dynamic-rel
-- Building x64-linux-dynamic-rel
-- Installing x64-linux-dynamic-rel
CMake Error at ports/x264/portfile.cmake: 51 (file):
  file RENAME failed to rename

  /path/to/vcpkg/packages/x264_64-linux-dynamic/lib/libx264.dll.lib

  to

  /path/to/vcpkg/packages/x264_64-linux-dynamic/lib/libx264.lib

  because: no such file or directory

Error: Building package x264:x64-linux-dynamic failed with: BUILD_FAILED
```

The problem is the following lines in the port file:

```
if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
    file(RENAME ${CURRENT_PACKAGES_DIR}/lib/libx264.dll.lib ${CURRENT_PACKAGES_DIR}/lib/libx264.lib)
    file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/libx264.dll.lib ${CURRENT_PACKAGES_DIR}/debug/lib/libx264.lib)
else() ...
```

It seems that this renaming operation should only be applicable on Windows, and if so then there should be an additional test within this if() statement.

### PR DESCRIPTION
**Pull request by JackBoosY**
https://github.com/microsoft/vcpkg/blob/5d33bd3d7910c6090911f71f9dcf1a4eecd303da/ports/x264/portfile.cmake#L50
Since x264 support linux, this condition is incorrect.

Fixes #13683.

### CODE REVIEW

**Review by LilyWangL — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]
LGTM once merged with master

File: ports/x264/CONTROL [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Source: x264
 Version: 157-303c484ec828ed0
-Port-Version: 9
+Port-Version: 10
```
Comment (JackBoosY) [inline ports/x264/CONTROL]:
```suggestion
Port-Version: 11
```

**Comment by JackBoosY** [comment]
Blocked by #13722.

**Comment by JackBoosY** [comment]
blocked by #13765.

**Pull request merged**